DAQ_System/Source/main.cpp

```diff
@@ -1,23 +1,140 @@
+                averageOffsetCalc(offset, sum);
+                offsetStatus = true; // offset is set so this setting makes it so it wont callibrate every single time.
+            }
+                
+
+            degreesPerSecondConversion(degreesPerSecond, offset, gyroData);
+            radiansPerSecondConversion(radiansPerSecond, degreesPerSecond);
+
+            /* this is the ANSI escape sequences 
+            \033 is the escape character, which signals the start of an escape sequence. 
+            [2J is the clear screen command, which clears the entire screen.
+            [1;1H is the cursor position command, which moves the cursor to a specified     
+            */
+                   
+            //raw data from the gyroscope
+            //printf("\033[2J\033[1;1H"); 
+            // printf("tx: %d \n",gyroData[0] ); 
+            //printf("ty: %d \n",gyroData[1] );
+            //printf("tz: %d \n",gyroData[2] );
+            // displays the average offset used for callibration of the gyroscope
+            //printf("the average of the X-axis is: %d \n", averageOffset[0]);
+            //printf("the average of the y-axis is: %d \n", averageOffset[1]);
+            //printf("the average of the z-axis is: %d \n", averageOffset[2]);
+            // displays the callibrated data of the gyroscope in dps
+            //printf("callibration of X(in dps): %d \n",degreesPerSecond[0]);
+            //printf("callibration of Y(in dps): %d \n",degreesPerSecond[1]);
+            //printf("callibration of Z(in dps): %d \n",degreesPerSecond[2]);
+            // displays the callibrated data of the gyroscope in radians per second
+            //printf("callibration of x(in rad/s) %.5f \n",radiansPerSecond[0]);
+            //printf("callibration of y(in rad/s) %.5f \n",radiansPerSecond[1]);
+            //printf("callibration of z(in rad/s) %.5f \n",radiansPersecond[2]);
+            
+  
+                    
+
+            //thread_sleep_for(500);
+
+                
+
+        } 
+    }     
+            
+}
+
+
+
+void averageOffsetCalc(short Offset[], short sum[])
+{
+  // collecting the sampled data for the offset average
+  short gyroData[3] = {0};
+  int averageSampleSize = 15;
+
+    for (int i = 0; i <= averageSampleSize; i++) 
+    {
+        gyro.read(gyroData);
+        sum[0] += gyroData[0];
+        sum[1] += gyroData[1];
+        sum[2] += gyroData[2];
+        // the if statment is used to find the average and will be calculated when all sample data is retrieved
+        if( i == averageSampleSize )
+        {
+            Offset[0] = sum[0]/averageSampleSize;
+            Offset[1] = sum[1]/averageSampleSize;
+            Offset[2] = sum[2]/averageSampleSize;
+        }
     }
-}
\ No newline at end of file
+
+}
+
+void degreesPerSecondConversion(short degreesPerSecond[],short offset[],short gyroData[])
+{
+            // the equation used to find the dps of the gyroscope is dps = (rawdata - offset)*sensitivity
```

Comment:
Can you document where the equations were derived and what resources you used?

Reply:
https://studylibfr.com/doc/10064916/lab-3---gyro-and-acc-lab--1-. 
 On page 6 of 10, that is where I got the equation.  One difference is that the number that is used for the sensitivity is different in my code than in this lab or whatever this is. The reason why I am using a different value is that on the datasheet for the l3gd20, it states that the default value for the gyroscope is 245 DPS(kinda weird since other gyroscope resources state that it is 250 DPS). But when I was looking at the Adafruit l3g20 library and the .cpp file (https://github.com/adafruit/Adafruit_L3GD20/blob/master/Adafruit_L3GD20.cpp) I found that they define it as 250 DPS (lines 192 - 195). So imma just go with the 250 DPS. The Adafruit library's calculations are also similar but they do not subtract any offsets to their readings.

here is a picture of the equation used from the first resource

![gyroequation](https://user-images.githubusercontent.com/113472670/221005539-03f68b91-a15e-4038-a4c5-66c35f8795d7.png)


here is a snippet of the Adafruit picture and how how they did the conversion
![gyroconversion](https://user-images.githubusercontent.com/113472670/221003695-6d08d24d-e06a-4785-bef2-72464747a63d.png)

also here is a picture of the data sheet where I found the default value for the range of the gyroscope. I'm not sure but maybe since there are two versions of the gyroscope( L3GD20H AND L3GD20) that is why I'm getting conflicting data for the DPS.

![gyroconversion](https://user-images.githubusercontent.com/113472670/221003969-4bc02a27-93df-4e28-aaee-e29b04b75ac6.png)

I also forgot to mention that for the sensitivity I used .00875 instead of the 8.75 value given in the datasheet. That is because the value given in the data sheet is in miliDPS instead of DPS

![sensitivity](https://user-images.githubusercontent.com/113472670/221006101-51252d27-66a1-426d-8b61-efddf389df4d.png)

Reply:
Ah, I see. You may have to make your own modifications to our version of the gyro. If you feel hesitant about a product-specific value, I suggest comparing their data sheets to see where they are derived from.

Anyway, make sure to add a comment in your code of the link and page number that you got it from. This will be really helpful for when someone else reads your code.

Reply:
Great job on keeping up with the sensor's documentation!